source/kraken/realtime.cpp

```diff
@@ -57,7 +60,69 @@ static void cancel_vj(type::MetaVehicleJourney* meta_vj,
     }
 }
 
-void handle_realtime(const transit_realtime::TripUpdate& trip_update, const type::Data& data) {
+static boost::shared_ptr<nt::new_disruption::Severity>
+make_no_service_severity(const boost::posix_time::ptime& timestamp,
+                         nt::new_disruption::DisruptionHolder& holder) {
+    // Yeah, that's quite hardcodded...
+    const std::string id = "kraken:rt:no_service";
+    auto& weak_severity = holder.severities[id];
+    if (auto severity = weak_severity.lock()) { return std::move(severity); }
+
+    auto severity = boost::make_shared<nt::new_disruption::Severity>();
+    severity->uri = id;
+    severity->wording = "trip canceled!";
+    severity->created_at = timestamp;
+    severity->updated_at = timestamp;
+    severity->color = "#000000";
+    severity->priority = 42;
+    severity->effect = nt::new_disruption::Effect::NO_SERVICE;
```

Comment:
If you have a good proposition to factorize this code easily, OK, but else, I let it like that.